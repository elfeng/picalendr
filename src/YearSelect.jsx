import React, { Component } from 'react';

const DEFAULT_YEAR = 0;

export default class YearSelect extends Component {

    constructor(props) {
        super(props);
        this.selectYear = this.selectYear.bind(this);
    }

    selectYear(yearNumber) {
        if (yearNumber !== DEFAULT_YEAR) {
            this.props.onChange(yearNumber.target.value);
        }
    }

    render() {

        const yearOptions = [2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017].map(year =>
            <option key={year} value={year}>{year}</option>
        );

        return (
            <select onChange={this.selectYear} className="form-control">
                <option value={DEFAULT_YEAR}>Year</option>
                {yearOptions}
            </select>
        );
    }
}; 