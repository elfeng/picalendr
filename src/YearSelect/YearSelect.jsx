import React, { Component } from 'react';

export default class YearSelect extends Component {

    constructor(props) {
        super(props);
        this.selectYear = this.selectYear.bind(this);
    }

    selectYear(yearNumber) {
        this.props.onChange(yearNumber.target.value);
    }

    render() {

        const yearOptions = [];
        for (let year = 2016; year >= 2000; year--){
            yearOptions.push(<option key={year} value={year}>{year}</option>);
        }

        return (
            <select onChange={this.selectYear} className="form-control" required
                id="year">
                <option value="">Year</option>
                {yearOptions}
            </select>
        );
    }
}; 