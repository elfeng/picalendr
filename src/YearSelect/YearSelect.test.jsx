import React from 'react';
import renderer from 'react-test-renderer';
import YearSelect from './YearSelect';

it('should match the snapshot', () => {
  const component = renderer.create(<YearSelect />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

