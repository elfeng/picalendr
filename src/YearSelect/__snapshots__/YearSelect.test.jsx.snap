exports[`test should match the snapshot 1`] = `
<select
  className="form-control"
  onChange={[Function]}
  required={true}>
  <option
    value="">
    Year
  </option>
  <option
    value={2010}>
    2010
  </option>
  <option
    value={2011}>
    2011
  </option>
  <option
    value={2012}>
    2012
  </option>
  <option
    value={2013}>
    2013
  </option>
  <option
    value={2014}>
    2014
  </option>
  <option
    value={2015}>
    2015
  </option>
  <option
    value={2016}>
    2016
  </option>
  <option
    value={2017}>
    2017
  </option>
</select>
`;
